### PR TITLE
Added Screenshot Functionality

### PR DESCRIPTION
**Pull request by shivam198smart**
Added Listener in Testng.xml
Added Code to capture screenshot if test gets fail

### CODE REVIEW

File: src/main/java/UITestFramework/CreateSession.java [hunk 1]
```diff
@@ -44,7 +47,30 @@
 	Properties configProp = new Properties();
 	String OS;
 
+	public WebDriver getDriver() {
+        return driver;
+}
+
+@BeforeMethod
```
Comment (shankybnl) [inline src/main/java/UITestFramework/CreateSession.java:54]: @shivam198smart There is already @BeforeMethod present. Any specific reason to add this new one? If not, please remove it.

File: src/main/java/UITestFramework/CreateSession.java [hunk 2]
```diff
@@ -44,7 +47,30 @@
...
+public void createDriver() {
+        WebDriver driver= getDriver();
+}
 
+@AfterMethod
```
Comment (shankybnl) [inline src/main/java/UITestFramework/CreateSession.java:59]: @shivam198smart @AfterMethod already present. Please add to it if you want anything new else I think we should remove it too.

File: src/main/java/UITestFramework/CreateSession.java [hunk 3]
```diff
@@ -44,7 +47,30 @@
 	Properties configProp = new Properties();
 	String OS;
 
+	public WebDriver getDriver() {
```
Comment (shankybnl) [inline src/main/java/UITestFramework/CreateSession.java:50]: Please add a single line comment on top of this method -

`/**
	 * Returns the instance of the webdriver.
	 * @return webdriver instance
	 */`

File: src/main/java/UITestFramework/retryLogic/TestListener.java [hunk 4]
```diff
@@ -28,15 +39,33 @@ public void onTestStart(ITestResult result) {
 	}
 
 	public void onTestSuccess(ITestResult result) {   }
-
-	public void onTestFailure(ITestResult result) {   }
+   
+	@Override
+	public void onTestFailure(ITestResult result) {
```
Comment (shankybnl) [inline src/main/java/UITestFramework/retryLogic/TestListener.java:44]: Please format the method lines.

File: testng.xml [hunk 5]
```diff
@@ -1,10 +1,13 @@
 <!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >
 <suite name="mobile-automation-framework">
-	<parameter name="os" value="${os}" />
-
+	<parameter name="os" value="android" />
```
Comment (shankybnl) [inline testng.xml:3]: Keep the `value="${os}"` as value is passed dynamically either android or ios

File: testng.xml [hunk 6]
```diff
@@ -1,10 +1,13 @@
...
+	<listeners>
+	<listener class-name="UITestFramework.retryLogic.TestListener"/> 
+	</listeners>
 	<test name="automation-tests">
+	<parameter name="os" value="android" />
```
Comment (shankybnl) [inline testng.xml:8]: I am do not think, this line is required

**Review by shankybnl — CHANGES_REQUESTED** [review]
Thank you for your contribution @shivam198smart . Please see the comments and make the required changes. Thanks again.

File: src/main/java/UITestFramework/CreateSession.java (hunk 1 above)
Comment (shivam198smart) [reply src/main/java/UITestFramework/CreateSession.java]: Removed.

File: src/main/java/UITestFramework/CreateSession.java (hunk 2 above)
Comment (shivam198smart) [reply src/main/java/UITestFramework/CreateSession.java]: Removed. It wasn't necessary.

File: src/main/java/UITestFramework/CreateSession.java (hunk 3 above)
Comment (shivam198smart) [reply src/main/java/UITestFramework/CreateSession.java:53]: Added the comment.

File: src/main/java/UITestFramework/retryLogic/TestListener.java (hunk 4 above)
Comment (shivam198smart) [reply src/main/java/UITestFramework/retryLogic/TestListener.java:42]: Formatted.

File: testng.xml (hunk 6 above)
Comment (shivam198smart) [reply testng.xml]: Removed

File: src/main/java/UITestFramework/retryLogic/TestListener.java [hunk 7]
```diff
@@ -34,36 +34,35 @@ public void onFinish(ITestContext context) {
...
-    File dest = new File("E:\\" + filename); //Directory where Screenshot get saved.
-   try {
-	FileUtils.copyFile(scr, dest);
-} catch (IOException e) {
-	// TODO Auto-generated catch block
-	e.printStackTrace();
-	}	
+			File scr = ((TakesScreenshot)webDriver).getScreenshotAs(OutputType.FILE);		
+			String filename =  new SimpleDateFormat("yyyyMMddhhmmss'.jpg'").format(new Date());
+    			File dest = new File("E:\\" + filename); //Directory where Screenshot get saved.
```
Comment (shankybnl) [inline src/main/java/UITestFramework/retryLogic/TestListener.java:49]: Please provide the `dest` path relative to the framework so it works on all the systems. Please try the below code and see if it creates screenshots at the mentioned destination. Thanks.

`File dest = new File("./Screenshots/" + filename);`

**Review by shankybnl — CHANGES_REQUESTED** [review]
Thank you for making the changes. Please check one review comment and make a change accordingly.

**Review by shankybnl — APPROVED** [review]

**Comment by shankybnl** [comment]
Thank you very much, @shivam198smart, for your contribution.

**Pull request merged**